include/tvm/auto_scheduler/measure.h

```diff
@@ -340,9 +351,14 @@ class LocalBuilderNode : public ProgramBuilderNode {
  public:
   /*! \brief Build function. */
   String build_func;
+  /*! \brief Functor with python implementation of submit method. */
+  PackedFunc submit_func;
```

Comment:
How about to pass a wrapped build function as the build_func like:
```python
wrapped_build_func(original_build_func, ...) {
    ....
    pass the original_build_func to your submit logic
    ....
}
```
Then we can avoid to add another strange API here.

Reply:
The main reason of using PackedFunc here instead of string with function name is next. I have to hold a ThreadPool object which will execute submitted tasks. Worker ThreadPool is not a singleton, it's an essential part of Builder/Runner instance and cannot be covered by `@tvm._ffi.register_func` mechanics.

In other words `submit_func` is not just static function, that's a holder for python based resources objects and methods.

I've tried to explain a little bit more detailed in [post](https://github.com/apache/tvm/pull/7877#issuecomment-822442102)